src/components/SSOLanding/useSSOSession.js

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useStore } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import queryString from 'query-string';
+
+import { config, okapi } from 'stripes-config';
+
+import { defaultErrors } from '../../constants';
+import { setAuthError } from '../../okapiActions';
+import { requestUserWithPerms } from '../../loginServices';
+import { parseJWT } from '../../helpers';
+
+const getParams = (location) => {
+  const search = location.search;
+
+  if (!search) return undefined;
+
+  return queryString.parse(search) || {};
+};
+
+const getToken = (cookies, params) => {
+  return cookies?.ssoToken || params?.ssoToken;
+};
+
+const getTenant = (params, token) => {
+  const tenant = config.useSecureTokens
+    ? params?.tenantId
+    : parseJWT(token)?.tenant;
+
+  return tenant || okapi.tenant;
+};
+
+const useSSOSession = () => {
+  const [isFailed, setIsFailed] = useState(false);
+  const store = useStore();
+  const dispatch = useDispatch();
+
+  const location = useLocation();
+  const [cookies] = useCookies(['ssoToken']);
+
+  const params = getParams(location);
+
+  const token = getToken(cookies, params);
```

Comment:
@NikitaSedyx and @zburke I think this means that this code still assumes that a token can be obtained from the cookie or the querystring, which would be true if we're running in "legacy mode" only not in RTR-land. So I think this means that https://folio-org.atlassian.net/browse/STCOR-764 isn't going to be resolved by this PR unless I'm missing something.

Reply:
@steveellis, you're correct that this function will return undefined under some circumstances, but that's fine. We could include checks, like the on lines 27-29, all over the place, but since it's an optional argument in the functions we pass it to, i.e. those functions already contain logic like `if (!token) {...}`, there is no value including that logic again here.

Reply:
it's correct that we support legacy and RTR approaches, but they now have common validation - _self response, if it fails so we redirect user to logic as there is no active session

I fully removed validation of tokens, query params etc, so we can support both authentication mechanisms (I had no goal to remove legacy as stripes supports it) with the same source of truth to know that session is valid or not

if valid token is provided in query params or httpOnly=false cookie - we set up UI session
if valid AT is provided in cookies - we set up UI session
if invalid token is provided in query params or httpOnly=false cookie - redirect to login with error
if invalid AT is provided in cookies (or not provided at all) - redirect to login with error

_self request says us that we can set up session or not

Reply:
Ok great guys, makes sense. Then I think this is fine. When we merge this it can be more easily tested by @renatsafiulin in https://folio-org.atlassian.net/browse/RANCHER-1250.

Reply:
I closed https://folio-org.atlassian.net/browse/STCOR-764 based on this.